openedx/core/djangoapps/schedules/models.py

```diff
@@ -5,10 +5,18 @@
 
 class Schedule(TimeStampedModel):
     enrollment = models.OneToOneField('student.CourseEnrollment', null=False)
-    active = models.BooleanField(default=True, help_text=_('Indicates if this schedule is actively used'))
-    start = models.DateTimeField(help_text=_('Date this schedule went into effect'))
+    active = models.BooleanField(
+        db_index=True,
```

Comment:
We want schedules whose deadlines are within a range and active, as well as schedules whose start time is within a range and active.

However, I think at the DB level we could just use the time index and then do the active filtering in python. The active flag isn't used at the moment.